core/include/mutex.h

```diff
@@ -249,7 +249,17 @@ static inline int mutex_trylock(mutex_t *mutex)
  *
  * @post    The mutex @p is locked and held by the calling thread.
  */
+#if defined(MODULE_CORE_MUTEX) || DOXYGEN
 void mutex_lock(mutex_t *mutex);
+#else
+static inline void mutex_lock(mutex_t *mutex)
+{
+    volatile void *lock = &mutex->queue.next;
+
+    while (*(volatile uintptr_t *)lock) {}
+    mutex->queue.next = &mutex->queue;
+}
+#endif
```

Comment:
This needs some commenting and explaining. IMO the intention is not obvious to provide a dummy implementation for use when no scheduler is present.

Reply:
Yea 
```C
volatile void *lock = &mutex->queue.next;

while (*(volatile uintptr_t *)lock) {}
mutex->queue.next = (uintptr_t)1;
```

is probably easier here.